### PR TITLE
Nrf52833 support

### PR DESCRIPTION
**Pull request by LokiMetaSmith**
adds nordic demo boards pca10059 and pca10100, also includes support for nrf52833

### CODE REVIEW

**Comment by hathach** [comment]
Sorry, I have been busy with other works, will review this whenever I could

File: variants/circuitplayground_nrf52840/variant.h [hunk 1]
```diff
@@ -70,12 +70,12 @@ extern "C"
...
-#define PIN_A3               (10)
-#define PIN_A4               (3)
-#define PIN_A5               (2)
-#define PIN_A6               (0)
+#define PIN_A1               (15)
+#define PIN_A2               (16)
+#define PIN_A3               (17)
+#define PIN_A4               (18)
+#define PIN_A5               (19)
+#define PIN_A6               (20)
```
Comment (hathach) [inline variants/circuitplayground_nrf52840/variant.h:78]: are you making these changes accidentally.

**Review by hathach — CHANGES_REQUESTED** [review]
I think you make accidental changes to following, therefore please revert it
- submodule TinyUSB_Arduino
- submodule nRFcrypto
- changes to cpb nrf52840

I will review it more once https://github.com/adafruit/Adafruit_nRF52_Arduino/pull/658 is merged, since it seems to have some conflict in platform.txt

File: boards.txt [hunk 2]
```diff
@@ -398,6 +398,94 @@ metro52840.menu.debug.l3.build.sysview_flags=-DCFG_SYSVIEW=1
...
+op1.menu.debug.l1.build.debug_flags=-DCFG_DEBUG=1
+op1.menu.debug.l2=Level 2 (Full Debug)
+op1.menu.debug.l2.build.debug_flags=-DCFG_DEBUG=2
+op1.menu.debug.l3=Level 3 (Segger SystemView)
+op1.menu.debug.l3.build.debug_flags=-DCFG_DEBUG=3
+op1.menu.debug.l3.build.sysview_flags=-DCFG_SYSVIEW=1
+
+
+# ----------------------------------
+# Generic NRF52833
```
Comment (hathach) [inline boards.txt:447]: generic board is not accepted, this should be changed to pca10100 if that is what you mean.

File: boards.txt [hunk 3]
```diff
@@ -398,6 +398,94 @@ metro52840.menu.debug.l3.build.sysview_flags=-DCFG_SYSVIEW=1
...
+op1.build.f_cpu=64000000
+op1.build.board=NRF52833_op1
+op1.build.core=nRF5
+op1.build.variant=opossum_pouch_nrf52833
+op1.build.usb_manufacturer="Nordic"
+op1.build.usb_product="nRF52833"
+op1.build.extra_flags=-DNRF52833_XXAA {build.flags.usb}
+op1.build.ldscript=nrf52833_s140_v6.ld
+op1.build.vid=0x239A
+op1.build.pid=0x8029
```
Comment (hathach) [inline boards.txt:426]: can you tell us more about this board. Also each board must has its own dedicated VID/PID and cannot re-use PID from another especially one is selling as commercial product.

**Review by hathach — CHANGES_REQUESTED** [review]
Most of the modification looks good, though in additional to previous review, please
- clarify the Opossum Pouch and its vid/pid 
- change generic board to pca10100
- remove the pca10059 (usb dongle) since the board requires soldering swd and flashing bootloader before actually using with this repo. Which is not really arduino friendly and could raise lots of support ticket from user.

File: variants/circuitplayground_nrf52840/variant.h (hunk 1 above)
Comment (LokiMetaSmith) [reply variants/circuitplayground_nrf52840/variant.h:78]: I might have pulled those from another commit. I didn't intend to change those.

File: boards.txt (hunk 3 above)
Comment (LokiMetaSmith) [reply boards.txt:426]: That's a mistake, Opossum Pouch is the board that I'm developing. I've got it defined on a private fork.

File: boards.txt (hunk 2 above)
Comment (LokiMetaSmith) [reply boards.txt:447]: Yes, that's the intent. I was reworking another commit and missed the documentation name change.

**Comment by hathach** [comment]
52833 is supported by https://github.com/adafruit/Adafruit_nRF52_Arduino/pull/739. Thank you for your patient

**Pull request closed without merging**